ui/src/dashboards/components/dashboard_index/DashboardCards.tsx

```diff
@@ -28,15 +26,110 @@ export default class DashboardCards extends PureComponent<Props> {
     getSortedResources
   )
 
+  state = {
+    hasMeasured: false,
+    pages: 1,
+    windowSize: 0,
+  }
+
+  public componentDidMount() {
```

Comment:
If the state is being set initially, why are we using componentDidMount to set it again?

Reply:
it's for when the component gets unmounted (like if you type in a filter that has no matches), and then remounted (remove the filter), to restart the measuring cycle

Reply:
Very interesting. I'll allow it